src/qiboml/models/decoding.py

```diff
@@ -283,3 +286,46 @@ def output_shape(self) -> tuple[int, int]:
     @property
     def analytic(self) -> bool:  # pragma: no cover
         return False
+    
+@dataclass
+class VariationalQuantumLinearSolver(QuantumDecoding):
+    """Decoder for the Variational Quantum Linear Solver (VQLS).
+    Adapted from the following paper arXiv:1909.05820v4 by Carlos Bravo-Prieto et al. 
+
+    Args:
+        target_state (ndarray): Target solution vector :math:`\\ket{b}`.
+        A (ndarray): The matrix ``A`` in the linear system :math:`A \\, \\ket{x} = \\ket{b}`.
+    
+    Ensure QiboML backend set to Pytorch. 
+    """
+    target_state: ndarray = None
+    A: ndarray = None
```

Comment:
```suggestion
    target_state: ndarray
    A: ndarray
```
are these actually optional? because you raise the error below when they are missing